internal/services/springcloud/spring_cloud_dev_tool_portal_resource.go

```diff
@@ -0,0 +1,404 @@
+package springcloud
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/springcloud/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/springcloud/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+	"github.com/tombuildsstuff/kermit/sdk/appplatform/2022-11-01-preview/appplatform"
+)
+
+type SpringCloudDevToolPortalModel struct {
+	Name                       string         `tfschema:"name"`
+	SpringCloudServiceId       string         `tfschema:"spring_cloud_service_id"`
+	Feature                    []FeatureModel `tfschema:"feature"`
+	PublicNetworkAccessEnabled bool           `tfschema:"public_network_access_enabled"`
+	Sso                        []SsoModel     `tfschema:"sso"`
+}
+
+type FeatureModel struct {
+	ApplicationAccelerator []FeatureDetailModel `tfschema:"application_accelerator"`
+	ApplicationLiveView    []FeatureDetailModel `tfschema:"application_live_view"`
+}
+
+type FeatureDetailModel struct {
+	Enabled bool `tfschema:"enabled"`
+}
+
+type SsoModel struct {
+	ClientId     string   `tfschema:"client_id"`
+	ClientSecret string   `tfschema:"client_secret"`
+	MetadataUrl  string   `tfschema:"metadata_url"`
+	Scope        []string `tfschema:"scope"`
+}
+
+type SpringCloudDevToolPortalResource struct{}
+
+var _ sdk.ResourceWithUpdate = SpringCloudDevToolPortalResource{}
+
+func (s SpringCloudDevToolPortalResource) ResourceType() string {
+	return "azurerm_spring_cloud_dev_tool_portal"
+}
+
+func (s SpringCloudDevToolPortalResource) ModelObject() interface{} {
+	return &SpringCloudDevToolPortalModel{}
+}
+
+func (s SpringCloudDevToolPortalResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return validate.SpringCloudDevToolPortalID
+}
+
+func (s SpringCloudDevToolPortalResource) Arguments() map[string]*schema.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringInSlice([]string{"default"}, false),
+		},
+
+		"spring_cloud_service_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.SpringCloudServiceID,
+		},
+
+		"feature": {
```

Comment:
i don't think we need to put these into a block called features? could we bring these up to the top level